fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java

```diff
@@ -1317,4 +1326,112 @@ public TSetConfigResponse setConfig(TSetConfigRequest request) throws TException
+                        tList.add(tableMetaInfo);
+                    });
+                } finally {
+                    db.readUnlock();
+                }                
+            }            
+        });
+        resp.tables_meta_infos = tList;
+        return resp;
+    }
+
+    static final ImmutableList<String> SHOW_PROPERTIES_NAME = new ImmutableList.Builder<String>()
+            .add(PropertyAnalyzer.PROPERTIES_STORAGE_TYPE)
+            .add(PropertyAnalyzer.PROPERTIES_STORAGE_MEDIUM)
+            .add(PropertyAnalyzer.PROPERTIES_STORAGE_COLDOWN_TIME)
+            .build();
+
+    private TTableMetaInfo genNormalTableMetaInfo(Table table, TTableMetaInfo tableMetaInfo) {
+        OlapTable olapTable = (OlapTable) table;
+        // Distribution info
+        DistributionInfo distributionInfo = olapTable.getDefaultDistributionInfo();
+        String distributeKey = distributionInfo.getDistributionKey();
+        String distributeNum = String.valueOf(distributionInfo.getBucketNum());
+        // Partition info
+        PartitionInfo partitionInfo = olapTable.getPartitionInfo();
+        StringBuilder partitionKeySb = new StringBuilder();
+        int idx = 0;
+        try {
+            for (Column column : partitionInfo.getPartitionColumns()) {
+                if (idx != 0) {
+                    partitionKeySb.append(", ");
+                }
+                partitionKeySb.append("`").append(column.getName()).append("`");
+                idx++;
+            }
+        } catch (NotImplementedException e) {
+            // This exception will be throwed except partition type is range
+            partitionKeySb.append("NULL");
+        }
+        tableMetaInfo.setPrimary_key(olapTable.getKeysType().equals(KeysType.PRIMARY_KEYS) ? distributeKey : "NULL");
+        tableMetaInfo.setPartition_key(partitionKeySb.toString());
+        tableMetaInfo.setDistribute_bucket(distributeNum);
+        tableMetaInfo.setDistribute_type("HASH");
+        tableMetaInfo.setDistribute_key(distributeKey);
+        tableMetaInfo.setSort_key(olapTable.getKeysType().toSql().split(" ")[0]);
+
+        Map<String, String> properties = olapTable.getTableProperty().getProperties();
+        Map<String, String> showProperties = new HashMap<>();
+        SHOW_PROPERTIES_NAME.forEach(key -> {
+            if (properties.containsKey(key)) {
+                showProperties.put(key, properties.get(key));
+            }
+        });
+        if (null != olapTable.getColocateGroup()) {
+            showProperties.put(PropertyAnalyzer.PROPERTIES_COLOCATE_WITH, olapTable.getColocateGroup());
+        }
+        Short replicationNum = olapTable.getDefaultReplicationNum();
+        showProperties.put(PropertyAnalyzer.PROPERTIES_REPLICATION_NUM, String.valueOf(replicationNum));
+        tableMetaInfo.setProperties(showProperties.toString());
+        return tableMetaInfo;
+    }
+
+    private TTableMetaInfo genDefaultMetaInfo(TTableMetaInfo tableMetaInfo) {
```

Comment:
"def" seems like a bug. I suggest you use "<unknown>" or "<not supported>".

Reply:
Mysql is using “def”, so I think it's ok

Reply:
ok then